### ISSUE TITLE
"native" instruction set alias for AOT compilers

### ISSUE THREAD

**Issue by jkotas** [dotnet/runtime#73246]
It would match the native architecture of the processor on which publishing happens.

Context:
- https://github.com/dotnet/runtime/pull/71911
- https://github.com/dotnet/BenchmarkDotNet/issues/2060

**Comment by MichalStrehovsky** [comment]
I think the most maintainable way might be to extract the CPU flag detection from the runtime:

https://github.com/dotnet/runtime/blob/cdf21f143735b8d104c8e636a37eb068904cdd8b/src/coreclr/nativeaot/Runtime/startup.cpp#L162-L305

Into a place that can be shared with the JitInterface native library:

https://github.com/dotnet/runtime/tree/cdf21f143735b8d104c8e636a37eb068904cdd8b/src/coreclr/tools/aot/jitinterface

Then compile that into jitinterface.dll (that ships with ILC) and p/invoke into this.

We already have managed definitions of the various flags this returns because the computed values are bitmasked with compile-time expectations burned into the produced executable to ensure we don't run on machines that don't have expected CPU features.

As a stretch goal, we might try to unify this detection with what's in CoreCLR VM, but that might be too much extra scope. Extracting something that would be eligible to be placed under src/native/minipal in the repo would be a very good first step towards that.

**Comment by JamesNK** [comment]
Performance hit was noticed when testing Native AOT gRPC app on Linux ARM.

AOT vs CoreCLR:
![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/303201/220266814-4bc0960d-1c84-42c4-a89f-16e5d66f6209.png)

Compared to a minor perf hit of AOT on Linux Intel:
![MicrosoftTeams-image (4)](https://user-images.githubusercontent.com/303201/220267039-a0b0fff5-0276-40e5-b0be-fafa3ccae6b7.png)

Probably culprit is EventSource methods that use Interlocked to increment longs:
https://github.com/grpc/grpc-dotnet/blob/0b365bf4633c9f05d0af374ed8607c046e8e74dd/src/Grpc.AspNetCore.Server/Internal/GrpcEventSource.cs#L67-L75

**Comment by EgorBo** [comment]
@JamesNK that makes sense, NativeAOT uses arm64 8.0 as a baseline while atomic instructions require 8.1, so you need to define `lse` capability for NativeAOT, e.g.: `<IlcInstructionSet>lse</IlcInstructionSet>`

I think a while ago we discussed about a named instruction set for Azure (to include the baseline instructions)

**Comment by JamesNK** [comment]
Yes, that fixed it.

Before: 239,492 RPS
After: 849,835 RPS

Also, only using Interlocked when required with this gRPC PR - https://github.com/grpc/grpc-dotnet/pull/2052 - will improve performance in the benchmark.

**Comment by omariom** [comment]
@JamesNK  If the effect this large then may be the hottest counters should be placed on their own cache lines?
